### PR TITLE
trilinos: add version 13.4.1

### PR DESCRIPTION
**Pull request by balay**
https://github.com/xsdk-project/xsdk-issues/issues/177#issuecomment-1274695246

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py
Comment (sethrj) [inline var/spack/repos/builtin/packages/trilinos/package.py]: Can you add the provenance for the patch file? Better yet, if the patch is a standalone PR/commit, can you delete the patch file and replace with something like:
```
    patch(
        "https://patch-diff.githubusercontent.com/raw/trilinos/Trilinos/pull/10545.patch?full_index=1",
        sha256="blah",
        when="@13.4.0 +superlu-dist"
    )
```

Comment (balay) [reply var/spack/repos/builtin/packages/trilinos/package.py]: Pushed this change.

I see the PR on the Trilinos side  isn't merged yet.

Perhaps this one should wait until then - and then use the URL to the merged commit..

**Comment by balay** [comment]
switching from `draft` to `don't-merge-yet` so that CI can be run

@spackbot run pipeline

**Comment by balay** [comment]
@sethrj I think this is ready-for-merge now

File: var/spack/repos/builtin/packages/trilinos/package.py
Comment (balay) [reply var/spack/repos/builtin/packages/trilinos/package.py]: So the above url is now a vaild url - so resolving this thread.

**Comment by balay** [comment]
Since `13.4.1` release is imminent - labeling this PR as `don't-merge-yet`

So we can use  `13.4.1` instead of applying this patch to  `13.4.0`

**Comment by sethrj** [comment]
OK so the plan is to mark up through 14.0.0 as "conflicts with " superlu 8, and then allow newer versions?

**Comment by balay** [comment]
> OK so the plan is to mark up through 14.0.0 as "conflicts with " superlu 8, and then allow newer versions?

Hm - I was planning on updating this PR to the following change - after `13.4.1` is spun:
```patch
diff --git a/var/spack/repos/builtin/packages/trilinos/package.py b/var/spack/repos/builtin/packages/trilinos/package.py
index 80765939cc..29345084ee 100644
--- a/var/spack/repos/builtin/packages/trilinos/package.py
+++ b/var/spack/repos/builtin/packages/trilinos/package.py
@@ -41,6 +41,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
 
     version("master", branch="master")
     version("develop", branch="develop")
+    version("13.4.1", sha256="0000000000000000000000000000000000000000000000000000000000000000")
     version("13.4.0", sha256="39550006e059043b7e2177f10467ae2f77fe639901aee91cbc1e359516ff8d3e")
     version("13.2.0", sha256="0ddb47784ba7b8a6b9a07a4822b33be508feb4ccd54301b2a5d10c9e54524b90")
     version(
@@ -421,7 +422,8 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("superlu-dist@:4.3", when="@11.14.1:12.6.1+superlu-dist")
     depends_on("superlu-dist@4.4:5.3", when="@12.6.2:12.12.1+superlu-dist")
     depends_on("superlu-dist@5.4:6.2.0", when="@12.12.2:13.0.0+superlu-dist")
-    depends_on("superlu-dist@6.3.0:7", when="@13.0.1:13 +superlu-dist")
+    depends_on("superlu-dist@6.3.0:7", when="@13.0.1:13.4.0 +superlu-dist")
+    depends_on("superlu-dist@8:", when="@13.4.1:13 +superlu-dist")
     depends_on("superlu-dist@develop", when="@master: +superlu-dist")
 
     # ###################### Patches ##########################
```

**Comment by sethrj** [comment]
@balay I think your first dependency is right but the second is too restrictive: from the trilinos PR older versions of SuperLU are still compatible. Maybe
```python
depends_on("superlu-dist@6.3.0:", when="@13.4.1:13 +superlu-dist")
```
?

**Comment by balay** [comment]
> depends_on("superlu-dist@6.3.0:", when="@13.4.1:13 +superlu-dist")

ok - will use this change

**Comment by balay** [comment]
@sethrj I think this PR is ready for approval/merge

**Review by sethrj — APPROVED** [review]

**Pull request merged**
